include/jxy/memory.hpp

```diff
@@ -89,6 +89,12 @@ class allocator
         return memory;
     }
 
+    template <typename Other>
+    struct rebind
```

Comment:
rebind was deprecated in cpp17 and removed in cpp20 I'm a little weary to accept this change because of this, allow me some time to investigate

Reply:
Okay, so I did a little digging and while `rebind` is deprecated on `std::allocator` it may be still required for this custom allocator:

https://en.cppreference.com/w/cpp/named_req/Allocator#cite_note-2

```
↑ rebind is only optional (provided by std::allocator_traits) if this allocator is a template of the form SomeAllocator<T, Args>, where Args is zero or more additional template type parameters.
```

That said, I'm a little perplexed why `std::_Replace_first_parameter<_Other,_Ty>` is tripping over this:

```
3>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xmemory(469,66): error C2027: use of undefined type 'std::_Replace_first_parameter<_Other,_Ty>'
3>        with
3>        [
3>            _Other=int,
3>            _Ty=jxy::details::allocator<int,PagedPool,809976148>
3>        ]
```

Let's at least comment/document the reason for this change so when someone goes to look at this later they understand why `jxy::allocator` has `rebind` functionality that doesn't mirror `std::allocator` deprecating it.